### PR TITLE
Color picker fixes

### LINKED ISSUE #4396: sliders of Color Picker behave wrong in RAW mode
there is a button named `raw mode` in color picker,
when turned on, rgb become from 0 to 1 but not from 0 to 255.
bug1: the slider still has the range of 0~255. it should be 0~1 then
bug2: alpha is not affected, at least in preview, e.g. 128 is still half opaque

### PR DESCRIPTION
**Pull request by djrm**
- Alpha is properly adjusted on raw mode fixes #4396
- Alpha to the end of the hex color string

### CODE REVIEW

File: core/color.cpp [hunk 1]
```diff
@@ -360,7 +360,7 @@ String Color::to_html(bool p_alpha) const {
 	txt+=_to_hex(g);
 	txt+=_to_hex(b);
 	if (p_alpha)
-		txt=_to_hex(a)+txt;
+		txt = txt + _to_hex(a);
```
Comment (akien-mga) [inline core/color.cpp]: Then `txt += _to_hex(a);` would be simpler no? (and like the increments above)

Comment (djrm) [inline core/color.cpp]: Yes will fix it later

**Comment by djrm** [comment]
Originally RAW mode clamps values to [0, 1] range, the changes here does just that, it clamps all the values (including alpha) to [0,1] range when raw mode is enabled, and multiplies those values by 255 when not in RAW mode, so if this is not right then what is RAW mode?

**Comment by djrm** [comment]
So raw mode range is [0,255]
And not raw is [0,1]
Or both are [0,255]
???

**Comment by reduz** [comment]
Raw is floats and raw mode from 0 to 1 equals normal from 0 to 255. Maybe
raw should be renamed to float

**Comment by vnen** [comment]
I get that raw mode can be greater than 1. But does it need to go all the way up to 255?

**Comment by reduz** [comment]
Could be as high as 100, but it's very useful to use high values to tint
sprites or objects (like tint a en enemy red when it takes damage or white
when about to explode)

**Comment by MarianoGnu** [comment]
I sugest to use 0% to 100% on normal mode and 0% to 1000% in raw mode, this will make it clear.
In both cases max alpha is 100%

**Comment by djrm** [comment]
ok so i understand the concept of raw mode, but the interface is confusing, because the value 255, can be confused with the an hex value, that is commonly used to represent colors, so i suggest to use 300, other thing is when is press raw mode in the current branch the values get normalized to [0,1], except for the alpha, and the slider does not change its limits, if raw mode is used to allow values grater than one and not-raw mode should not allow values greater than one then sliders in not-raw mode should have a max value of 1, which is not the case (currently the max value is always 255) in both modes (i mean trying to select a value).

**Comment by djrm** [comment]
also max alpha should be always one not 255, if we are using color ranges from 0  to 1

**Comment by MarianoGnu** [comment]
In non raw mode max values are 1, there's no inconsistence there, the mistake was having alpha range changed

**Comment by djrm** [comment]
ok, so in code one can get the color (that is normalized), and in the interface the values are represented as hex components (maybe the misunderstanding was here).

screenshot from 2016-07-02 17-58-47

so 255 is one. im going to think that this is just a convenience for the user.
Then in raw mode the color values can be higher than 1, specifically as high as 255, maybe the documentation should clarify that Raw mode is not color8 and that it is just a coincidence (maybe i will work on that).

Maybe the highest value in Raw mode should be configurable by the user via a method, like
set_max_value_for_raw_mode (just a suggestion).

**Comment by MarianoGnu** [comment]
@djrm the # is a button, you can press it to get a Color(x,x,x,x) constructor from the line editor :)

**Pull request closed without merging**